app/src/main/java/com/travelcy/travelcy/ui/split/SplitViewModel.kt

```diff
@@ -23,14 +32,48 @@ class SplitViewModel(private val billRepository: BillRepository, private val cur
         }
 
         addSource(foreignCurrency) {
-           recalculateTotal(billItemsWithPersons.value, it)
+            recalculateTotal(billItemsWithPersons.value, it)
         }
 
         addSource(billItemsWithPersons) {
             recalculateTotal(it, foreignCurrency.value)
         }
     }
 
+    val totalAmountPerPerson: MediatorLiveData<List<Pair<Person, String>>> = MediatorLiveData<List<Pair<Person, String>>>().apply {
+        fun recalculateAmountPerPerson(persons: List<Person>?, billItemsWithPersons: List<BillItemWithPersons>?, foreign: Currency?)  {
+            val pricesPerPerson = persons?.map { person ->
+                val priceForPerson = billItemsWithPersons?.sumByDouble { billItemWithPersons ->
+                    val hasPerson = billItemWithPersons.persons.find { it.id == person.id } != null
+                    if (hasPerson) {
+                        billItemWithPersons.billItem.amount * billItemWithPersons.billItem.quantity
```

Comment:
Possible missing division by number of persons assigned to the bill item?

Reply:
You are correct! Well spotted, I'll fix this!

Reply:
This is fixed in e2543e9, I also made the code more testable and added tests for this, as well as the total amount, to ensure this works correcly in the future. 00ac4ce then fixes an issue with the locale in the tests that I added and ec53252 fixes various issues with the add currencies modal